### ISSUE TITLE
API keys should be organizational resource

### ISSUE THREAD

**Issue by prabhatsharma** [openobserve/openobserve#1089]
### Which OpenObserve functionalities are relevant/related to the feature request?

ingestion

### Description

API keys are currently tied to a user in the organization. If the user leaves the organization and her account is deleted then the token for ingestion will stop working and impact the whole organization.

### Proposed solution

ingestion tookens are currently user based resource. tokens should instead be organization level resource like dashboard or stream. There should be a separate section for defining API keys / tokens

### Alternatives considered

na

**Comment by oasisk** [comment]
RBAC will have impact on API keys/tokens.

**Comment by mr-karan** [comment]
Marking my interest in this. I think an ingestion token where a role "ingest" (which can only insert data) is really required. It'll be awesome if the token can be scoped to a stream only (N streams: N tokens).

**Comment by gaby** [comment]
> Marking my interest in this. I think an ingestion token where a role "ingest" (which can only insert data) is really required. It'll be awesome if the token can be scoped to a stream only (N streams: N tokens).

YES! This is the way Elastic/Kibana do it. The token only allows ingesting data, nothing else.
